### PR TITLE
[OrderByComparator] LPD-27018

### PR DESCRIPTION
**Pull request by victorhcunha**
[LPD-27018](https://liferay.atlassian.net/browse/LPD-27018)

[LPD-27018]: https://liferay.atlassian.net/browse/LPD-27018?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:report:57679381

**Comment by victorhcunha** [comment]
ci:report:57617936

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:test:upgrade

**Comment by victorhcunha** [comment]
ci:report:59482564

**Comment by victorhcunha** [comment]
ci:report:59443856

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:test:upgrade

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:report:59970455

**Comment by victorhcunha** [comment]
ci:report:59965380

File: modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/util/comparator/SegmentsExperimentModifiedDateComparator.java [hunk 1]
```diff
@@ -23,12 +23,14 @@ public class SegmentsExperimentModifiedDateComparator
 
 	public static final String[] ORDER_BY_FIELDS = {"modifiedDate"};
 
-	public SegmentsExperimentModifiedDateComparator() {
-		this(false);
-	}
+	public static SegmentsExperimentModifiedDateComparator getInstance(
```
Comment (dantewang) [inline modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/util/comparator/SegmentsExperimentModifiedDateComparator.java:26]: @victorhcunha This is an internal class and is not used, you can remove it.

File: modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjava [hunk 2]
```diff
@@ -205,6 +209,14 @@ public class UpgradeCatchAllCheck implements ConfigurationBeanDeclaration {
 		FolderArticleTitleComparator.getInstance(false);
 		FolderIdComparator.getInstance(false);
 		FolderNameComparator.getInstance(false);
+		FragmentCollectionCreateDateComparator.getInstance(false);
```
Comment (dantewang) [inline modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjava:212]: @victorhcunha This one should be `true`.

File: modules/util/source-formatter/src/main/resources/dependencies/replacements.json [hunk 3]
```diff
@@ -1048,6 +1048,162 @@
...
+		"from": "new ExportImportConfigurationNameComparator(boolean paramName)",
+		"issueKey": "LPD-27018",
+		"skipParametersValidation": true,
+		"to": "ExportImportConfigurationNameComparator.getInstance(param#0#)"
+	},
+	{
+		"from": "new FragmentCollectionCreateDateComparator()",
+		"issueKey": "LPD-27018",
+		"skipParametersValidation": true,
+		"to": "FragmentCollectionCreateDateComparator.getInstance(false)"
```
Comment (dantewang) [inline modules/util/source-formatter/src/main/resources/dependencies/replacements.json:1103]: @victorhcunha This one should be `true`.

**Review by dantewang — CHANGES_REQUESTED** [review]
@victorhcunha Please see my comment and update the PR, thanks!

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
Thanks @dantewang ! I updated the pr with the changes

**Comment by victorhcunha** [comment]
ci:report:61152624

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:stop

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
ci:report:2024-08-13

**Comment by victorhcunha** [comment]
ci:report:62007850

**Comment by victorhcunha** [comment]
ci:test:upgrade

**Comment by victorhcunha** [comment]
ci:stop

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:report:62533843

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
ci:stop

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:test:upgrade

**Comment by victorhcunha** [comment]
ci:report:63923743

**Comment by victorhcunha** [comment]
ci:report:63913493

**Comment by dantewang** [comment]
ci:forward:force

**Pull request closed without merging**